lib/events.js

```diff
@@ -70,6 +70,23 @@ async function create(pg_pool, req, res /* regex */) {
       app.space_name,
       `at=error dynos="${Array.isArray(payload.dynos) ? payload.dynos.map((x) => x.type).join(',') : 'unknown'}" code=${payload.code} desc="${payload.description}" restarts=${payload.restarts}`,
     );
+  } else if (payload.action === 'security_scan') {
+    // Make sure the payload conforms to the expected format
+    // Must include status, service, and message. Can optionally include a link.
+    if (!payload.status || !payload.service_name || !payload.message) {
+      throw new common.BadRequestError('Payload must include status, service_name, and message.');
+    }
+    // Remove anything that isn't status, service, message, link, app info, etc.
+    payload = {
+      action: payload.action,
+      key: payload.key,
+      app: payload.app,
+      space: payload.space,
+      status: payload.status,
+      service_name: payload.service,
+      message: payload.message,
+      link: payload.link || undefined,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unexpected use of undefined.](https://app.codacy.com/manual/Akkeris/controller-api/pullRequest?prid=5256669)